files/api/user/services/grant.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+// Public node modules.
+const _ = require('lodash');
+const Purest = require('purest');
+
+/**
+ * Connect thanks to a third-party provider.
+ *
+ *
+ * @param {String}    provider
+ * @param {String}    access_token
+ *
+ * @return  {*}
+ */
+
+exports.connect = function * connect(provider, access_token) {
+  const deferred = Promise.defer();
+
+  if (!access_token) {
+    deferred.reject({
+      message: 'No access_token.'
+    });
+  } else {
+    // Get the profile.
+    getProfile(provider, access_token, function (err, profile) {
+      if (err) {
+        deferred.reject(err);
+      } else {
+        // We need at least the mail.
+        if (!profile.email) {
+          deferred.reject({
+            message: 'Email was not available.'
+          });
+        } else {
+          User.findOne({email: profile.email}).exec(function (err, user) {
+            if (err) {
+              deferred.reject(err);
+            } else if (!user) {
+              // Create the new user.
+              const params = _.assign(profile, {
+                id_ref: 1,
+                lang: strapi.config.i18n.defaultLocale,
+                template: 'standard',
+                provider: provider
+              });
+
+              User.create(params).exec(function (err, user) {
+                if (err) {
+                  deferred.reject(err);
+                } else {
+                  deferred.resolve(user);
+                }
+              });
+            } else {
+              deferred.resolve(user);
+            }
+          });
+        }
+      }
+    });
+  }
+
+  return deferred.promise;
+};
+
+/**
+ * Helper to get profiles
+ *
+ * @param {String}   provider
+ * @param {Function} callback
+ */
+
+function getProfile(provider, access_token, callback) {
+  switch (provider) {
+    case 'facebook':
+      const facebook = new Purest({provider: 'facebook'});
```

Comment:
The app does not start with Node v0.12.

![capture d ecran 2015-11-06 a 16 16 45](https://cloud.githubusercontent.com/assets/5550462/11000447/1d961b28-84a2-11e5-9e17-5e9636643377.png)